samples/snippets/src/main/java/com/example/dialogflow/AnswerRecordManagement.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.dialogflow;
+
+import java.io.IOException;
+import com.google.api.gax.rpc.ApiException;
+import com.google.cloud.dialogflow.v2.AnswerRecordsClient;
+import com.google.cloud.dialogflow.v2.UpdateAnswerRecordRequest;
+import com.google.cloud.dialogflow.v2.AnswerFeedback;
+import com.google.cloud.dialogflow.v2.AnswerRecord;
+import com.google.protobuf.FieldMask;
+public class AnswerRecordManagement {
+
+  // [START dialogflow_update_answer_record]
+  /**
+   * Update the answer record
+   * @param projectId The GCP project linked with the conversation profile.
+   * @param answerRecordId The answer record id returned along with the suggestion.
+   * @param clicked The boolean value indicating whether the answer record was clicked.
+   * @return updated answer record.
+   * @throws ApiException
+   * @throws IOException
+   */
+  /**
+   * 
+   * @param projectId
+   * @param answerRecordId
+   * @param clicked
+   * @return
+   * @throws ApiException
+   * @throws IOException
+   */
+  public static AnswerRecord updateAnswerRecord(
+      String projectId, 
+      String answerRecordId, 
+      boolean clicked)
+      throws ApiException, IOException {
+    try (AnswerRecordsClient answerRecordsClient = AnswerRecordsClient.create()) {
+      String answerRecordName = 
+        "projects/" + projectId + "/locations/global/answerRecords/" + answerRecordId;
```

Comment:
If there is a resource-name class available for use, we should use it rather than building resource name strings.

```suggestion
      AnswerRecordName answerRecordName = 
        AnswerRecordName.ofProjectLocationAnswerRecordName(projectId, "global", answerRecordId);
```

Reply:
The doc https://googleapis.dev/java/google-cloud-dialogflow/latest/com/google/cloud/dialogflow/v2/AnswerRecordName.html#ofProjectLocationAnswerRecordName-java.lang.String-java.lang.String-java.lang.String- says it's a BetaApi, should we still use it?

Reply:
That‘s possible. I have made this change.